bindings/src/stream_reader.rs

```diff
@@ -93,6 +86,23 @@ impl StreamReader {
         Ok(py_future_clone)
     }
 
+    ///
+    /// Set the reader offline.
+    ///
+    #[text_signature = "($self)"]
+    pub fn reader_offline(&self) -> PyResult<()> {
+        self.handle.block_on(self.reader_offline_async());
+        Ok(())
+    }
+
+    #[text_signature = "($self, slice)"]
+    pub fn release_segment(&self, slice: &mut Slice) -> PyResult<()> {
```

Comment:
Does this need to be public?

Reply:
I have implemented it as a public API. An example usage pattern is used in the test file `pravega_reader_test.py`
This could be extended to releasing a segment slice at an offset too. Let me know if this should be enabled too.